### PR TITLE
feat: add NaN detection during training

### LINKED ISSUE #4985: [Feature Request] throw an error when the total loss is NaN
### Summary

When the total loss is NaN, we can throw an error to stop the training. Otherwise, time is wasted training a model with parameters to be NaN, as seen in the case of https://github.com/deepmodeling/dpgen/issues/1460.

### Detailed Description

1. NaN can be checked when the total loss is on the CPU (but not the GPU), to avoid extra cost. For example, when writing to `lcurve.out`, the result is on the CPU.
2. Check the results before the checkpoint is written, so no checkpoint with NaN is written.
3. Implement the feature for TensorFlow, PyTorch, and PaddlePaddle backends.

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by njzjz**
Fix #4985.

This implementation is much simpler than #4986.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved training-metric validation to detect NaN total RMSE, logging a clear error and halting runs to avoid silent failures.
* **Documentation**
  * Added documentation for the new option that controls NaN checking so users can enable or disable the validation as needed.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: deepmd/loggers/training.py [hunk 1]
```diff
@@ -28,8 +48,16 @@ def format_training_message_per_task(
         lr = f", lr = {learning_rate:8.2e}"
     # sort rmse
     rmse = dict(sorted(rmse.items()))
-    return (
+    msg = (
         f"batch {batch:7d}: {task_name}"
         f"{', '.join([f'{kk} = {vv:8.2e}' for kk, vv in rmse.items()])}"
         f"{lr}"
     )
+    if check_total_rmse_nan and math.isnan(rmse.get("rmse", 0.0)):
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/loggers/training.py:56]: This is incorrect. `rmse` is directly set in the loss class. Other RMSEs may be NaN in expectation.

File: deepmd/loggers/training.py [hunk 2]
```diff
@@ -28,8 +48,16 @@ def format_training_message_per_task(
...
+        log.error(msg)
+        err_msg = (
+            f"NaN detected at batch {batch:7d}: {task_name}. "
+            "Something went wrong, and it is meaningless to continue."
+        )
+        raise RuntimeError(err_msg)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/loggers/training.py:62]: It may not be expected if some loss doesn't give the total rmse. Or we may consider always setting it in the training.

**Review by wanghan-iapcm — APPROVED** [review]

**Review by iProzd — APPROVED** [review]
